caddyconfig/httpcaddyfile/pkiapp.go

```diff
@@ -16,23 +16,106 @@ package httpcaddyfile
 
 import (
 	"github.com/caddyserver/caddy/v2/caddyconfig"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
 	"github.com/caddyserver/caddy/v2/modules/caddypki"
 )
 
+func init() {
+	RegisterGlobalOption("pki", parsePKIApp)
+}
+
+// parsePKIApp parses the global log option. Syntax:
+//
+//     pki [id] {
+//         name                     <name>
+//         root_common_name         <name>
+//         intermediate_common_name <name>
+//     }
+//
+// When the CA ID is unspecified, 'local' is assumed.
+//
+func parsePKIApp(d *caddyfile.Dispenser, existingVal interface{}) (interface{}, error) {
```

Comment:
Wonder if this would fit better in a new file called `caddyfile.go` or something, just to keep it out of the way of the main PKI app.

Reply:
Hmm? This code is in `httpcaddyfile` already

Reply:
Oh, true. Nevermind.